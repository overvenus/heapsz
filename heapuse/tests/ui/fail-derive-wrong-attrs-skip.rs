use heapuse_derive::Heap;

#[path = "../../examples/allowlisting.rs"]
mod allowlisting;

#[derive(Heap)]
pub struct WrongSkip1 {
    #[heap_size(skip)]
    pub a: allowlisting::FullAllow,
}

#[derive(Heap)]
pub enum WrongSkip2 {
    #[heap_size(skip)]
    A(usize),
    B(usize),
}

fn main() {}
