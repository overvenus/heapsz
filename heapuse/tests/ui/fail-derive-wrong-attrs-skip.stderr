error: `#[heap_size(skip)]` is only allow with a container attribute `#[heap_size]`.
 --> tests/ui/fail-derive-wrong-attrs-skip.rs:8:17
  |
8 |     #[heap_size(skip)]
  |                 ^^^^

error: `#[heap_size(skip)]` is only allow with a container attribute `#[heap_size]`.
  --> tests/ui/fail-derive-wrong-attrs-skip.rs:14:17
   |
14 |     #[heap_size(skip)]
   |                 ^^^^
